pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2205,6 +2391,128 @@ static __always_inline int save_args_to_submit_buf(event_data_t *data, u64 types
     return arg_num;
 }
 
+// INTERNAL: SYSCALLS ------------------------------------------------------------------------------
+
+static __always_inline bool has_syscall_fd_arg(uint syscall_id)
+{
+    // Only syscalls with one fd argument so far
+    switch (syscall_id) {
```

Comment:
Only way to "fix this long list" I can think of (because values are not sequential) would be to transform the syscall defines into a feature table (syscall, archnumber, features bit flags), but we can't have global variables so... its tricky to improve this :\

Reply:
I think we can't run away entirely from `#define`s. A draf suggestion would be to create a `get_syscall()`, to be called in sys_enter, which returns a bit flagged syscall structure based on arch and syscall_id?

```c
static __always_inline struct syscall get_syscall(uint syscall_id, uint arch)
{
	struct syscall s = {
		.syscall_id = syscall_id,
		.arch = arch,
	};

	switch (s.syscall_id) {
		case SYSCALL_READ:
			s.flags = FD_ARG1
			break;
		...
		case SYSCALL_DUP2:
		case SYSCALL_DUP3:
			s.flags = FD_ARG_1 | FD_ARG_2;
			break;
	}

	return s;
}
```

It would be a initial step for checking flags from specific use cases.